pipeline/ffmpeg.go

```diff
@@ -30,26 +30,36 @@ func (f *ffmpeg) HandleStartUploadJob(job *JobInfo) (*HandlerOutput, error) {
 		return nil, fmt.Errorf("cannot create sourceOutputUrl: %w", err)
 	}
 	sourceOutputURL := sourceOutputBaseURL.JoinPath(job.RequestID)
-	segmentingTargetURL := sourceOutputURL.JoinPath(SEGMENTING_SUBDIR, SEGMENTING_TARGET_MANIFEST)
-
 	job.SourceOutputURL = sourceOutputURL.String()
-	job.SegmentingTargetURL = segmentingTargetURL.String()
-	log.AddContext(job.RequestID, "segmented_url", job.SegmentingTargetURL)
-
-	// Begin Segmenting
-	log.Log(job.RequestID, "Beginning segmenting via FFMPEG/Livepeer pipeline")
-	job.ReportProgress(clients.TranscodeStatusPreparing, 0.5)
 
-	// FFMPEG fails when presented with a raw IP + Path type URL, so we prepend "http://" to it
-	internalAddress := config.HTTPInternalAddress
-	if !strings.HasPrefix(internalAddress, "http") {
-		internalAddress = "http://" + internalAddress
-	}
+	segmentingTargetURL := sourceOutputURL.JoinPath(SEGMENTING_SUBDIR, SEGMENTING_TARGET_MANIFEST)
+	job.SegmentingTargetURL = segmentingTargetURL.String()
 
-	destinationURL := fmt.Sprintf("%s/api/ffmpeg/%s/index.m3u8", internalAddress, job.StreamName)
-	if err := video.Segment(job.SignedSourceURL, destinationURL, job.TargetSegmentSizeSecs); err != nil {
-		return nil, err
+	// Only do the Segmenting step for non-hls input files
+	if job.InputFileInfo.Format != "hls" {
+
+		log.Log(job.RequestID, "Beginning segmenting via FFMPEG/Livepeer pipeline")
+		job.ReportProgress(clients.TranscodeStatusPreparing, 0.5)
+
+		// FFMPEG fails when presented with a raw IP + Path type URL, so we prepend "http://" to it
+		internalAddress := config.HTTPInternalAddress
+		if !strings.HasPrefix(internalAddress, "http") {
+			internalAddress = "http://" + internalAddress
+		}
+
+		destinationURL := fmt.Sprintf("%s/api/ffmpeg/%s/index.m3u8", internalAddress, job.StreamName)
+		if err := video.Segment(job.SignedSourceURL, destinationURL, job.TargetSegmentSizeSecs); err != nil {
+			return nil, err
+		}
+	} else {
+		// If hls input is detected (i.e. a recording), override the SegmentingTargetURL
+		// to use the SourceFile manifest since segmentation step is skipped.
+		// And also make it an OS compatible URL so that later stages can download it
+		if strings.HasPrefix(job.SourceFile, "http") {
+			job.SegmentingTargetURL = "s3+" + job.SourceFile
+		}
```

Comment:
I'm not convinced that this is the best place to do this. It feels a bit error prone...